mlflow/pyfunc/scoring_server/__init__.py

```diff
@@ -185,6 +190,13 @@ def infer_and_parse_data(data, schema: Schema = None):
     """
 
     format_keys = set(data.keys()).intersection(SUPPORTED_FORMATS)
+
+    # If the input doesn't contain any keys in the supported formats,
+    # then we assume that the client wants to treat the input as-is
+    # and we skip any additional processing
+    if len(format_keys) == 0:
+        return data
+
```

Comment:
I think this skips signature enforcement, as the logic happens inside `parse_tf_serving_input` or `dataframe_from_parsed_json`.

Reply:
Correction: This skips data parsing logic, signature enforcement still happens inside predict. Wondering if it's not consistent with other input format.

Reply:
Is there anything important going on in those parsing functions that should apply to dicts (e.g. filtering or something)? From what I saw, it looks like it was mainly creating the appropriate `np.arrays` and `dtypes`. If the user wants to just treat their input as a dict, would schema enforcement inside `predict` be enough?